src/ensembl/src/shared/select/Select.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useState, useRef, useEffect } from 'react';
+import classNames from 'classnames';
+
+import { splitFromSelected } from './helpers/select-helpers';
+import * as keyCodes from 'src/shared/constants/keyCodes';
+
+import SelectOptionsPanel from './SelectOptionsPanel';
+import SelectArrowhead from './SelectArrowhead';
+
+import styles from './Select.scss';
+
+export type Option = {
+  value: any;
+  label: React.ReactNode;
+  isSelected: boolean;
+  isDisabled?: boolean;
+};
+
+export type OptionGroup = {
+  title?: string;
+  options: Option[];
+};
+
+export type GroupedOptionIndex = [
+  number, // index of option group
+  number // index of the option within a group
+];
+
+type ClosedSelectProps = {
+  isOpen: boolean;
+  selectedOption: Option | null;
+  placeholder: string;
+  onClick: () => void;
+};
+
+type OptionsSpecificProps = {
+  options: Option[];
+  title?: string;
+};
+
+type OptionGroupsSpecificProps = {
+  optionGroups: OptionGroup[];
+};
+
+type CommonProps = {
+  onSelect: (value: any) => void;
+  placeholder?: string;
+};
+
+type OptionsSelectProps = CommonProps & OptionsSpecificProps;
+type OptionGroupsSelectProps = CommonProps & OptionGroupsSpecificProps;
+
+type SelectAdapterProps = OptionsSelectProps | OptionGroupsSelectProps;
+
+type SelectProps = OptionGroupsSelectProps & { placeholder: string };
+
+// TODO: think of a better name for this component?
+const ClosedSelect = (props: ClosedSelectProps) => {
```

Comment:
Something like SelectController or SelectHolder ?

Reply:
I checked what they call this component in React-Select, and it is called `Control` there:

![image](https://user-images.githubusercontent.com/6834224/56595780-2d196300-65e7-11e9-95aa-6fd947d01661.png)

I am not sure I like that name, but it may be better than ClosedSelect.